Source/WebCore/loader/cache/MemoryCache.h

```diff
@@ -168,7 +168,7 @@ class MemoryCache {
     WEBCORE_EXPORT void pruneLiveResourcesToSize(unsigned targetSize, bool shouldDestroyDecodedDataForAllLiveResources = false);
 
 private:
-    using CachedResourceMap = HashMap<std::pair<URL, String /* partitionName */>, CachedResource*>;
+    using CachedResourceMap = HashMap<std::pair<URL, String /* partitionName */>, WeakPtr<CachedResource>>;
```

Comment:
Since we now hold a WeakPtr, rather than a bare pointer, we avoid ever using one of these after it is deleted. But I don't think that absolves us of needing to destroy the underlying resource, if we used to be responsible through the `::remove()` method. Who does that deletion now?

Reply:
See my earlier comment. This change is about not using a pointer after free (security hardening), not about deletion.